### PR TITLE
ticket-1620 IMA event log scanner is now in-place code, external module was removed

### LINKED ISSUE #1620: IMA: getPastEvents sometimes work significantly slow as expected
**Description** 
Case: compatibility check 
IMA version 11/16: 2.0.0
IMA version 5/16: 1.5.0-beta.1
Skaled: the same for all - 3.16.1

**Step to reproduce**
Run transfers loop between schain
When processing message takes more than 60 seconds observe logs on working node

**Actual result** 
In some regular use case search of IMA msg can takes more then 30-... min, depends on amount of blocks on the schain. 
NOTE: Root cause investigate in progress 

**Expected result** 
IMA processing msg should not takes more than 40 seconds in average. getPastEvents should parse logs not longer than 5-7 seconds based on previous results, and results on nodes not affected by this problem.

NOTE: Look closer on logs, attention to PWA thread  on affected chain. 
Compare imaGPE_big.txt with imaIterativeScan13.txt and imaIterativeScan12.txt

[imaGPE_big.txt](https://github.com/skalenetwork/IMA/files/13022989/imaGPE_big.txt)
[imaGPE.txt](https://github.com/skalenetwork/IMA/files/13022986/imaGPE.txt)
[imaIterativeScan12.txt](https://github.com/skalenetwork/IMA/files/13022987/imaIterativeScan12.txt)
[imaIS12.txt](https://github.com/skalenetwork/IMA/files/13022988/imaIS12.txt)
[imaIterativeScan13.txt](https://github.com/skalenetwork/IMA/files/13022983/imaIterativeScan13.txt)
[imaGPE13.txt](https://github.com/skalenetwork/IMA/files/13022984/imaGPE13.txt)

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1554
- [ ] https://github.com/skalenetwork/IMA/issues/1554
```

```[tasklist]
### Tasks
- [ ] https://github.com/skalenetwork/IMA/issues/1554
```

### PR DESCRIPTION
**Pull request by sergiy-skalelabs**
IMA event log scanner is now in-place code, external module was removed. No new tests needed. We must re-check it in legacy network also.

### CODE REVIEW

File: npms/skale-ima/index.mjs
Comment (DmytroNazarenko) [inline npms/skale-ima/index.mjs:1]: Why haven't you reverted these lines: 
https://github.com/skalenetwork/IMA/compare/develop...revert-1623-ticket-1620/resolve-speed-issues-of-IMA-event-scanner?expand=1#diff-ad8baf7ead6c78572c88576791e9f1b1f78c9b9932595b191931a649c5fa6115L357-L361

https://github.com/skalenetwork/IMA/compare/develop...revert-1623-ticket-1620/resolve-speed-issues-of-IMA-event-scanner?expand=1#diff-ad8baf7ead6c78572c88576791e9f1b1f78c9b9932595b191931a649c5fa6115L925-L928 

https://github.com/skalenetwork/IMA/compare/develop...revert-1623-ticket-1620/resolve-speed-issues-of-IMA-event-scanner?expand=1#diff-ad8baf7ead6c78572c88576791e9f1b1f78c9b9932595b191931a649c5fa6115L950-L954 

https://github.com/skalenetwork/IMA/compare/develop...revert-1623-ticket-1620/resolve-speed-issues-of-IMA-event-scanner?expand=1#diff-ad8baf7ead6c78572c88576791e9f1b1f78c9b9932595b191931a649c5fa6115L1266-L1267

File: npms/skale-ima/imaEventLogScan.mjs [hunk 1]
```diff
@@ -188,6 +127,14 @@ export async function safeGetPastEventsProgressive(
...
         );
     }
+    if( log.verboseGet() >= log.verboseReversed().information ) {
+        details.write( strLogPrefix +
+            cc.info( "Will run progressive event log search for event " ) +
+            cc.j( strEventName ) + cc.info( " via URL " ) +
+            cc.u( owaspUtils.ethersProviderToUrl( ethersProvider ) ) +
+            generateWhileTransferringLogMessageSuffix( optsChainPair ) +
+            cc.info( "..." ) );
+    }
```
Comment (DmytroNazarenko) [inline npms/skale-ima/imaEventLogScan.mjs:130]: What is the purpose of this log? There was no such log in reverted PR

**Review by DmytroNazarenko — CHANGES_REQUESTED** [review]
Why these files are not reverted:
agent/loop.mjs
agent/clpTools.mjs
agent/test/agentUnitTests.js

https://github.com/skalenetwork/IMA/compare/develop...revert-1623-ticket-1620/resolve-speed-issues-of-IMA-event-scanner?expand=1#

File: npms/skale-ima/imaEventLogScan.mjs (hunk 1 above)
Comment (sergiy-skalelabs) [reply npms/skale-ima/imaEventLogScan.mjs:130]: This is requested by Evg - print node number we requesting

File: npms/skale-ima/index.mjs
Comment (sergiy-skalelabs) [reply npms/skale-ima/index.mjs:1]: Links does not open source code

**Comment by DmytroNazarenko** [comment]
https://github.com/skalenetwork/IMA/pull/1627

**Pull request closed without merging**